plus/oop/delete.js

```diff
@@ -6,4 +6,15 @@
  * @param {object} o the object
  * @param {string} key the name of the property to delete
  * @returns {object} the new object without the given property
- */
\ No newline at end of file
+ */
+
+function del (o, key) {    
+    if (typeof o !== "object") {
+        throw new Error ("No object is passed")
+    }
+
+    const newObject = Object.assign({}, o);
+    return delete newObject[key]
```

Comment:
delete operator returns the whether the deletion was successful or not, not the object without the property